contracts/token/ERC721/ERC721.sol

```diff
@@ -151,28 +152,23 @@ abstract contract ERC721 is Context, ERC165, IERC721, IERC721Metadata, IERC721Er
     /**
      * @dev See {IERC721-transferFrom}.
      */
-    function transferFrom(address from, address to, uint256 tokenId) public virtual {
-        if (!_isApprovedOrOwner(_msgSender(), tokenId)) {
-            revert ERC721InsufficientApproval(_msgSender(), tokenId);
-        }
-
+    function transferFrom(address from, address to, uint256 tokenId) public {
```

Comment:
Why are we removing virtual from these function ?

Reply:
Same reason for `_transfer` but I suppose the public functions have other considerations. Do you think they should remain virtual?

Reply:
yes.

Our guidelines are to make (almost) everything virtual. I'm ok to making `_transfer`, `_mint` and `_burn` non virtual, because they can be bypassed by calling `_update` directly. But apart from these 3, everything else should remain virtual IMO.